app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -215,6 +215,12 @@ class HomeFragment : Fragment() {
             sessionControlInteractor,
             homeViewModel
         )
+
+        // This has to be called separately from the consumeFrom block since the coroutine doesn't
+        // allow our UI to be updated right away and delays our start up. The init block allows us to
+        // force our UI to render with the data available in our store at fragment creation time.
+        sessionControlView?.update(homeFragmentStore.state)
```

Comment:
**Big issue**: what happens if the data isn't available when this is called? We might want to consider blocking on it to make this more deterministic (e.g. if our fnprms starts reporting two drastically different numbers randomly, we'll be confused).

I confirmed this isn't deterministic as is: when running debug builds on Pixel 2, I see the top sites animation.

Reply:
The list should be empty and will just update itself as its doing now. we could block on it too if needed